homeassistant/components/delijn/sensor.py

```diff
@@ -0,0 +1,107 @@
+"""Support for De Lijn (Flemish public transport) information."""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_NEXT_PASSAGE = 'nextpassage'
+CONF_STOP_ID = 'stop_id'
+CONF_API_KEY = 'api_key'
+CONF_MAX_PASSAGES = 'max_passages'
+
+DEFAULT_NAME = 'De Lijn'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_NEXT_PASSAGE): [{
+        vol.Required(CONF_STOP_ID): cv.string,
+        vol.Optional(CONF_MAX_PASSAGES, default=5): cv.positive_int}]
+})
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Create the sensor."""
+    from pydelijn.api import Passages
+
+    api_key = config.get(CONF_API_KEY)
+    name = DEFAULT_NAME
+
+    session = async_get_clientsession(hass)
+
+    sensors = []
+    for nextpassage in config.get(CONF_NEXT_PASSAGE):
+        stop_id = nextpassage[CONF_STOP_ID]
+        max_passages = nextpassage[CONF_MAX_PASSAGES]
+        line = Passages(hass.loop, stop_id, max_passages, api_key, session)
+        sensors.append(DeLijnPublicTransportSensor(line, name))
+
+    tasks = [sensor.async_update() for sensor in sensors]
+    if tasks:
+        await asyncio.wait(tasks)
+    if not all(sensor._attributes for sensor in sensors):
+        raise PlatformNotReady
+
+    async_add_entities(sensors, True)
+
+
+class DeLijnPublicTransportSensor(Entity):
+    """Representation of a Ruter sensor."""
+
+    def __init__(self, line, name):
+        """Initialize the sensor."""
+        self.line = line
+        self._attributes = {}
+        self._name = name
+        self._state = None
+
+    async def async_update(self):
+        """Get the latest data from the De Lijn API."""
+        await self.line.get_passages()
+        if self.line.passages is None:
+            _LOGGER.error("No data recieved from De Lijn")
+            return
+        try:
+            attributes = {}
+            first = self.line.passages[0]
+            self._state = first['due_in_min']
+            self._name = first['stopname']
+            attributes['stopname'] = first['stopname']
+            attributes['line_number_public'] = first['line_number_public']
+            attributes['line_transport_type'] = first['line_transport_type']
+            attributes['final_destination'] = first['final_destination']
+            attributes['due_at_sch'] = first['due_at_sch']
```

Comment:
What does `due_at_sch` mean?

Reply:
Sometimes the real-time data is not available (or not made available) by the API, for example when the tram/bus did not leave yet at the starting station there is not real-time info yet but only the scheduled info is available. That's why there's a split made between scheduled (due_at_sch) and real-time (due_at_rt) times.

Reply:
Can we write out `schedule` instead of `sch`, and real-time` instead of `rt`?

Reply:
Changed the due_at_sch to due_at_schedule and due_at_rt to due_at_realtime in the attributes[] but not in the first[] as those are the names used by the pydelijn package.